### PR TITLE
DEV: Populate table content using raw markdown

### PR DESCRIPTION
**Pull request by keegangeorge**
This PR handles two things.

(1) Firstly, rather than taking data from cooked text from the HTML table, we will now take data from the raw Markdown and build JS objects (after parsing with `MarkdownIt`'s `PrettyText` Engine) to feed the MD data into the Spreadsheet.

This resolves the important issue of things like links inside tables getting overridden with plaintext.

(2) Additionally, since the method is already being modified, logic was added to make the column width based on the the larger of the content and heading character length. Previously only the heading character length was considered.

This will thus check-off two items in the to do list:
- [x]  FIX: links and html ex. <small> inside tables are reduced to plain text after editing
- [X] UX: Make column width based on larger of content/heading

### CODE REVIEW

File: javascripts/discourse/api-initializers/table-editor.js [hunk 1]
```diff
@@ -35,12 +39,26 @@ export default apiInitializer("0.11.1", (api) => {
...
+            const tokens = prettyText.opts.engine.parse(post.raw, {
+              references: {},
+            });
+            const allTables = tokenRange(tokens, "table_open", "table_close");
+            const tableTokens = allTables[tableId];
+
+            showModal("insert-table-modal", {
+              model: post,
+            }).setProperties({
+              tableHtml: tempTable,
```
Comment (pmusaraj) [inline javascripts/discourse/api-initializers/table-editor.js:56]: This is no longer needed, right?

File: javascripts/discourse/components/spreadsheet-editor.js [hunk 2]
```diff
@@ -108,29 +108,46 @@ export default class SpreadsheetEditor extends GlimmerComponent {
...
+  buildPopulatedTable(tableTokens) {
+    const contentRows = tokenRange(tableTokens, "tr_open", "tr_close");
+    const rows = [];
+    let headings;
+    const rowWidthFactor = 8;
+
+    contentRows.forEach((row, index) => {
+      if (index === 0) {
+        // headings
+        headings = this.extractTableContent(row).map((heading) => {
```
Comment (pmusaraj) [inline javascripts/discourse/components/spreadsheet-editor.js:127]: This is ok for now, but can also refactor to use thead as an identifier for the table header. Not sure how a table without headings would be handled currently.

**Review by pmusaraj — APPROVED** [review]
Looks good, left some minor comments. 

Would be good to add a couple of tests in a followup PR, this is starting to really take shape.

File: javascripts/discourse/components/spreadsheet-editor.js (hunk 2 above)
Comment (keegangeorge) [reply javascripts/discourse/components/spreadsheet-editor.js:127]: Ah yes, I haven't tried a table without headings yet. I can test that next.

I was previously using the `<th>` to check, but I saw the headings appear when I was looking at the rows, so I thought I'd just compile it into one.

File: javascripts/discourse/api-initializers/table-editor.js (hunk 1 above)
Comment (keegangeorge) [reply javascripts/discourse/api-initializers/table-editor.js:56]: I am still using it here:
```
  get isEditingTable() {
    if (this.args.tableHtml) {
      return true;
    }

    return false;
  }
```

as I was having some issues when using `this.args.tableTokens` instead, but I'll be trying to get rid of it.

**Pull request merged**